### PR TITLE
python310Packages.matplotx: init at 0.3.10

### PR DESCRIPTION
**Pull request by swflint**
###### Description of changes

Package Matplotx, a Python library that provides a number of quality-of-life and quality-of-output features to users of Matplotlib.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

ciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

### CODE REVIEW

File: pkgs/development/python-modules/matplotx/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    rev = "v${version}";
+    hash = "sha256-EWEiEY23uFwd/vgWVLCH/buUmgRqz1rqqlJEdXINYMg=";
+  };
+
+  propagatedBuildInputs = [
+    matplotlib
+    numpy
+    networkx
+    pypng
+    scipy
```
Comment (natsukium) [inline pkgs/development/python-modules/matplotx/default.nix:29]: These packages are optional dependencies so they should be added to `passthru.optional-dependencies`.
See, [17.27.2.3.2. Optional extra dependencies](https://nixos.org/manual/nixpkgs/stable/#reference)

File: pkgs/development/python-modules/matplotx/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ];
+
+  pythonImportsCheck = [ "matplotx" ];
+
+  meta = {
+    homepage = "https://github.com/nschloe/matplotx";
+    description = "More styles and useful extensions for Matplotlib";
```
Comment (natsukium) [inline pkgs/development/python-modules/matplotx/default.nix:36]:
```suggestion
    description = "More styles and useful extensions for Matplotlib";
    changelog = "https://github.com/nschloe/matplotx/releases/tag/v${version}"
```
It is better to add a changelog to the meta.

**Review by natsukium — COMMENTED** [review]
`matplotx` is a python module, so the `python3Packags.matplotx` is preferred for a commit message.

File: pkgs/development/python-modules/matplotx/default.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    owner = "nschloe";
+    repo = "matplotx";
+    rev = "v${version}";
+    hash = "sha256-EWEiEY23uFwd/vgWVLCH/buUmgRqz1rqqlJEdXINYMg=";
+  };
+
+  propagatedBuildInputs = [
+    matplotlib
+    numpy
+
```
Comment (ocfox) [inline pkgs/development/python-modules/matplotx/default.nix:27]:
```suggestion
```

**Comment by natsukium** [comment]
It looks fine except that `imageio` and `pytest-cov` are not needed.

File: pkgs/development/python-modules/matplotx/default.nix [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+, fetchFromGitHub
+, pytestCheckHook
+, setuptools
+, matplotlib
+, numpy
+, networkx
+, pypng
+, scipy
+, imageio
+, pytest-cov
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/matplotx/default.nix:12]:
```suggestion
```

**Comment by SuperSandro2000** [comment]
@ofborg build python311Packages.matplotx

**Pull request merged**